pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesTest.java

```diff
@@ -439,6 +441,144 @@ public void testCheckTimeBasedBacklogQuota() throws Exception {
         admin.topics().deletePartitionedTopic(testTopic, true);
     }
 
+    @Test
+    public void testPriorityOfGlobalPolicies() throws Exception {
+        final SystemTopicBasedTopicPoliciesService topicPoliciesService
+                = (SystemTopicBasedTopicPoliciesService) pulsar.getTopicPoliciesService();
+        final JerseyClient httpClient = JerseyClientBuilder.createClient();
+        // create topic and load it up.
+        final String namespace = myNamespace;
+        final String topic = BrokerTestUtil.newUniqueName("persistent://" + namespace + "/tp");
+        final TopicName topicName = TopicName.get(topic);
+        final String hostAndPort = pulsar.getWebServiceAddress();
+        final String httpPath = "/admin/v2/persistent/" + namespace + "/" + TopicName.get(topic).getLocalName()
+                + "/maxConsumers";
+        admin.topics().createNonPartitionedTopic(topic);
+        Producer producer = pulsarClient.newProducer().topic(topic).create();
+        PersistentTopic persistentTopic =
+                (PersistentTopic) pulsar.getBrokerService().getTopics().get(topic).get().get();
+
+        // Set non global policy.
+        // Verify: it affects.
+        Response res = httpClient.target(hostAndPort).path(httpPath)
+                .queryParam("isGlobal", "false")
+                .request()
+                .header("Content-Type", "application/json")
+                .post(Entity.json(10));
+        assertTrue(res.getStatus() == 200 || res.getStatus() == 204);
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(topicPoliciesService.getTopicPoliciesAsync(topicName, LOCAL_ONLY).join().get()
+                    .getMaxConsumerPerTopic(), 10);
+            HierarchyTopicPolicies hierarchyTopicPolicies = persistentTopic.getHierarchyTopicPolicies();
+            assertEquals(hierarchyTopicPolicies.getMaxConsumerPerTopic().get(), 10);
+        });
+
+        // Set global policy.
+        // Verify: topic policies has higher priority than global policies.
+        Response globalRes = httpClient.target(hostAndPort).path(httpPath)
```

Comment:
Please use `admin.topicPolicies(true).setMaxConsumers(topic,20)` instead of `httpClient`.

Reply:
Added more test cases and covered the case of using pulsar-admin API

```java
    @DataProvider(name = "clientRequestType")
    public Object[][] clientRequestType() {
        return new Object[][]{
            {"PULSAR_ADMIN"},
            {"HTTP"}
        };
    }